contracts/registry/RegistryServiceReleaseManager.sol

```diff
@@ -0,0 +1,335 @@
+import {IService} from "../shared/IService.sol";
+import {ProxyManager} from "../shared/ProxyManager.sol";
+
+import {IRegistry} from "./IRegistry.sol";
+import {Registry} from "./Registry.sol";
+import {IRegistryService} from "./IRegistryService.sol";
+import {RegistryService} from "./RegistryService.sol";
+import {RegistryServiceManager} from "./RegistryServiceManager.sol";
+import {RegistryServiceAccessManager} from "./RegistryServiceAccessManager.sol";
+
+
+contract RegistryServiceReleaseManager is AccessManaged
+{
+    struct ReleaseInfo {
+        ObjectType[] types; // service types of release
+        uint awaitingRegistration;// "services left to register" counter 
+    }
+
+    // unique role for some service function
+    struct ConfigInfo {
+        bytes4[] selector;
+        RoleId roleId; 
+    }
+
+    RegistryServiceAccessManager private immutable _accessManager;
+    // TODO keep mapping of all proxy managers
+    RegistryServiceManager private immutable _proxyManager;
+    IRegistry private immutable _registry;
+
+    VersionPart _latest;// latest active version
+    VersionPart _initial;// first active version
+
+    mapping(VersionPart version => ReleaseInfo info) _release;
+
+    mapping(VersionPart version => mapping(ObjectType serviceType => ConfigInfo)) _config;
+
+    // TODO parametrize initial implementations?
+    constructor(
+        RegistryServiceAccessManager accessManager/*,
+        bytes memory registryByteCodeWithInitCode,
+        bytes memory registryServiceByetCodeWithInitCode*/)
+        AccessManaged(accessManager.authority())
+    {
+        _accessManager = accessManager;
+
+        address owner = msg.sender;
+        _proxyManager = new RegistryServiceManager(
+            owner,
+            address(this)
+            /*initialAuthority, 
+            type(Registry).creationCode, 
+            type(RegistryService).creationCode*/
+        );
+
+        IRegistryService registryService = _proxyManager.getRegistryService();
+        _registry = registryService.getRegistry();
+
+        // get initial release version
+        VersionPart initialVersion = registryService.getMajorVersion();
+        //require(initialVersion.toInt() > 0, "Release manager: initial version 0");
+        _initial = initialVersion;
+        //_latest = zeroVersionPart();// 0 - no activated releases yet
+        
+        // create initial release
+        //IRegistryService.FunctionConfig[] memory config = registryService.getConfig();
+        (, bytes memory data) = registryService.getInitialInfo();
+        IRegistryService.FunctionConfig[] memory config = abi.decode(data, (IRegistryService.FunctionConfig[]));
+
+        _verifyAndStoreConfig(address(registryService), config);
+    }
+
+    function createNextRelease(IService registryService)
+        external
+        restricted // GIF_ADMIN_ROLE
+        returns(NftId nftId)
+    {
+        // check interface
+        // check authority
+
```

Comment:
ensure the latest release is fully deployed.

Reply:
At the first call to createNextRelease() registry service is registered and _awaitingDeployment is set. Consequent calls will revert because nextVersion will point to the same value used in the first call. In order to increase next version value _awaitingDeployment must go to 0 ~ all service domains must get registered